rct229/rulesets/ashrae9012019/section1/section1rule1.py

```diff
@@ -0,0 +1,142 @@
+            ruleset_section_title="Performance Calculations",
+            standard_section="Section G4.2.1.1",
+            is_primary_rule=True,
+            list_path="ruleset_model_descriptions[0]",
+            data_items={"climate_zone": (BASELINE_0, "weather/climate_zone")},
+        )
+
+    class RMDRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section1Rule1.RMDRule, self).__init__(
+                rmrs_used=produce_ruleset_model_instance(
+                    USER=True,
+                    BASELINE_0=True,
+                    BASELINE_90=True,
+                    BASELINE_180=True,
+                    BASELINE_270=True,
+                    PROPOSED=True,
+                ),
+                rmrs_used_optional=produce_ruleset_model_instance(
+                    USER=True,
+                    BASELINE_0=True,
+                    BASELINE_90=True,
+                    BASELINE_180=True,
+                    BASELINE_270=True,
+                    PROPOSED=True,
+                ),
+                manual_check_required_msg=MANUAL_CHECK_REQUIRED_MSG,
+            )
+
+        def get_calc_vals(self, context, data=None):
+            rmd_u = context.USER
+            rmd_b0 = context.BASELINE_0
+            rmd_b90 = context.BASELINE_90
+            rmd_b180 = context.BASELINE_180
+            rmd_b270 = context.BASELINE_270
+            rmd_p = context.PROPOSED
+
+            output_bpf_set = []
+
+            for rmd in (rmd_u, rmd_b0, rmd_b90, rmd_b180, rmd_b270, rmd_p):
+                if rmd is not None:
+                    output_bpf_set.append(
+                        find_one(
+                            "$.output.total_area_weighted_building_performance_factor",
+                            rmd,
+                        )
+                    )
+
+            output_bpf_set = list(filter(lambda x: x is not None, output_bpf_set))
+            assert_(
+                len(output_bpf_set) >= 1,
+                "At least one `output_bpf_set` value must exist.",
+            )
+            bpf_building_area_type_dict = get_BPF_building_area_types_and_zones(rmd_b0)
+            is_undetermined = "UNDETERMINED" in bpf_building_area_type_dict.keys()
+            bpf_bat_sum_prod = 0
+            total_area = ZERO.AREA
+            climate_zone = data["climate_zone"]
+            for bpf_bat in bpf_building_area_type_dict.keys():
+                if is_undetermined:
+                    continue
+                expected_bpf = table_4_2_1_1_lookup(bpf_bat, climate_zone)[
+                    "building_performance_factor"
+                ]
+                total_area += bpf_building_area_type_dict[bpf_bat]["area"]
+                bpf_bat_sum_prod += (
+                    expected_bpf * bpf_building_area_type_dict[bpf_bat]["area"]
+                )
+            return {
```

Comment:
Since the ```total_area``` term goes to the denominator in the rule_check, please check if it's zero or not here as shown below.
```Python
assert_(
                total_area > 0,
                "The `total_area ` value must be greater than 0.",
            )
```